core/heap.c

```diff
@@ -1401,6 +1401,9 @@ vmm_heap_init()
     if (DYNAMO_OPTION(vm_reserve)) {
         vmm_heap_unit_init(&heapmgt->vmheap, DYNAMO_OPTION(vm_size));
     }
+    dynamo_vm_areas_lock();
+    heap_exiting = false;
```

Comment:
Are you suggesting just not grabbing the lock because there's only one thread during init? I think that's safe on x86 (acquire-release semantics by default), but wasn't sure about ARM.
Or you mean flipping the flags back once detach is over?